keyvault/deterministic/store.go

```diff
@@ -0,0 +1,67 @@
+/*
+Package deterministic defines an implementation of an insecure,
+purely deterministic keyvault used by this remote signer server
+for demonstration purposes.
+
+WARN: This is a toy implementation that returns deterministic BLS12-381
+secret+public key pairs, NOT MEANT FOR PRODUCTION.
+*/
+package deterministic
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/shared/bls"
+	"github.com/prysmaticlabs/prysm/shared/bytesutil"
+	"github.com/prysmaticlabs/prysm/shared/interop"
+	"github.com/sirupsen/logrus"
+)
+
+var log = logrus.WithField("prefix", "deterministic-keyvault")
+
+// Store defines a deterministic keyvault, written for demonstrative purposes.
+type Store struct {
+	pubKeysToSecretKeys map[[48]byte]bls.SecretKey
+	pubKeys             []bls.PublicKey
+}
+
+// NewStore instantiates a deterministic keyvault using a set number of keys.
+func NewStore(numKeys int) (*Store, error) {
+	log.Infof("Generating %d determistic keys...", numKeys)
+	secretKeys, pubKeys, err := interop.DeterministicallyGenerateKeys(
+		0 /* start idx */, uint64(numKeys),
+	)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Could not deterministically generate %d keys", numKeys)
+	}
+	pubKeysToSecretKeys := make(map[[48]byte]bls.SecretKey)
+	for i := 0; i < len(pubKeys); i++ {
+		pubKey := bytesutil.ToBytes48(pubKeys[i].Marshal())
+		pubKeysToSecretKeys[pubKey] = secretKeys[i]
+	}
+	log.WithField(
+		"numKeys", numKeys,
+	).Info("Initialized deterministic keyvault")
+	s := &Store{
+		pubKeys:             pubKeys,
+		pubKeysToSecretKeys: pubKeysToSecretKeys,
+	}
+	return s, nil
+}
+
+// GetSecretKey returns the corresponding secret key for a BLS12-381 public key.
+func (s *Store) GetSecretKey(ctx context.Context, pubKey bls.PublicKey) (bls.SecretKey, error) {
```

Comment:
Again, `ctx` is unused, not sure if that is intentional (to be used in further cleanup PR). The same for `GetPublicKeys()`

Reply:
It's important to always have ctx as the first argument in interface definitions as a best practice, such as:
```go
type Store interface {
  GetSecretKey(context.Context, bls.PublicKey) (bls.SecretKey, error)
}
```
As most implementations of the interface will wish to leverage context in one way or another

Reply:
For example, a `Store` implementation using something such as aws S3 or hashicorp vault might need the incoming context for their own purposes

Reply:
Yep, I also think that `ctx` is probably the only good case where you can add variable for future, not immediate use within a created function.